web/src/engine/websites/ComicK.ts

```diff
@@ -146,9 +150,42 @@ export default class extends DecoratableMangaScraper {
     }
 
     public override async FetchPages(chapter: Chapter): Promise<Page[]> {
-        const uri = new URL('/chapter/' + chapter.Identifier, this.apiUrl);
-        const request = new Request(uri.href);
-        const data = await FetchJSON<APIChapter>(request);
-        return data.chapter.md_images.map(image => new Page(this, chapter, new URL(`https://meo.comick.pictures/${image.b2key}`), { Referer: this.URI.href }));
+        const data = await FetchJSON<APISingleChapter>(new Request(new URL(`/chapter/${chapter.Identifier}`, this.apiUrl)));
+        return data.chapter.md_images.map(image => new Page(this, chapter, this.ComputePageURL(image, data.chapter), { Referer: this.URI.href }));
     }
+
+    private ComputePageURL(image: APIPage, chapter: APIChapter): URL {
+        let result = '';
+        if (image.gpurl && !image.b2key) image.gpurl.match(/siasky/) ? result = image.gpurl : result = `https://lh3.googleusercontent.com/${image.gpurl}`;
+
+        if (image.b2key) {
+            result = this.ComputePageURLFromB2key(new Date(chapter.created_at) > new Date('2024-03-14 06:00:00') ? chapter.id : 1, image.b2key);
+        } else {
+            const server = 'mangadex.org' == chapter.server ? 'images.comick.fun' : chapter.server;
+            result = chapter.chapter_id || (null == server ? void 0 : server.match(/mangadex/)) ? `https://'${server}/data/${chapter.hash}/${image.name}` : `https://${server}/${chapter.hash}/${image.name}`;
+        }
+        return new URL(result);
+    }
+
+    private ComputePageURLFromB2key(chapterid: number, b2key: string): string {
+        let CDN = 'https://meo2.comick.pictures/';
```

Comment:
I copied the code from the website, i'd rather stick to it 🤷 .

Reply:
Moreover, this method gives a better image quality than the javascript you suggested to use, and i assume better than with the tachiyomi parameter which i we dont use for the sake of quality

Reply:
Nice catch but : 

`let t = 'https://' + (0, s.po) ().replace('meo.comick.pictures', 's3.comick.ink/comick') + '/' + e,`. 

They replace only `meo.comick.pictures`, not the other two. 
So i dont see the reason to replace `meo`, `meo2`, and `meo3`.

Moreover,  `s3.comick.ink/comick` is used in website's download function, which is login-walled. For now we can access to s3 pictures but this may not last. I'd rater stick to the website's code, for maintenance's sake. And i think the quality is identical on both 🤷 .